flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1662,6 +1667,7 @@ private class TaskCanceler implements Runnable {
         @Override
         public void run() {
             try {
```

Comment:
This could be a good candidate for MdcUtils#withContext or `MDC.putCloseable()`
(more instances in this file)

Reply:
Replaced all instances.